go.mod

```diff
@@ -0,0 +1,168 @@
+	github.com/packer-community/winrmcp v0.0.0-20180921211025-c76d91c1e7db // indirect
+	github.com/pborman/uuid v1.2.0
+	github.com/pierrec/lz4 v2.3.0+incompatible // indirect
+	github.com/pkg/errors v0.9.1
+	github.com/pkg/sftp v1.10.1
+	github.com/prometheus/client_golang v1.2.1 // indirect
+	github.com/prometheus/client_model v0.0.0-20191202183732-d1d2010b5bee // indirect
+	github.com/prometheus/procfs v0.0.8 // indirect
+	github.com/satori/uuid v1.2.0 // indirect
+	github.com/shurcooL/httpfs v0.0.0-20190707220628-8d4bc4ba7749 // indirect
+	github.com/shurcooL/vfsgen v0.0.0-20181202132449-6a9ea43bcacd
+	github.com/sirupsen/logrus v1.4.2
+	github.com/smartystreets/goconvey v1.6.4 // indirect
+	github.com/spf13/cobra v0.0.5
+	github.com/stoewer/go-strcase v1.1.0 // indirect
+	github.com/stretchr/testify v1.4.0
+	github.com/terraform-providers/terraform-provider-aws v1.60.1-0.20191219195657-d7cd077ea2a8
+	github.com/terraform-providers/terraform-provider-azuread v0.7.0 // indirect
+	github.com/terraform-providers/terraform-provider-azurerm v1.41.0
+	github.com/terraform-providers/terraform-provider-google v1.20.1-0.20191217221207-85d4b5989b26
+	github.com/terraform-providers/terraform-provider-ignition v1.2.1
+	github.com/terraform-providers/terraform-provider-local v1.4.0
+	github.com/terraform-providers/terraform-provider-openstack v1.24.0
+	github.com/terraform-providers/terraform-provider-random v1.3.2-0.20191204175905-53436297444a
+	github.com/terraform-providers/terraform-provider-vsphere v1.14.0
+	github.com/vincent-petithory/dataurl v0.0.0-20191104211930-d1553a71de50
+	github.com/vmware/govmomi v0.22.1
+	go.uber.org/atomic v1.5.1 // indirect
+	go.uber.org/multierr v1.4.0 // indirect
+	go.uber.org/zap v1.13.0 // indirect
+	golang.org/x/crypto v0.0.0-20191206172530-e9b2fee46413
+	golang.org/x/lint v0.0.0-20191125180803-fdd1cda4f05f
+	golang.org/x/oauth2 v0.0.0-20191202225959-858c2ad4c8b6
+	golang.org/x/sys v0.0.0-20191206220618-eeba5f6aabab
+	golang.org/x/time v0.0.0-20191024005414-555d28b269f0 // indirect
+	golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543 // indirect
+	google.golang.org/api v0.14.0
+	gopkg.in/AlecAivazis/survey.v1 v1.8.7
+	gopkg.in/ini.v1 v1.51.0
+	gopkg.in/yaml.v2 v2.2.7
+	k8s.io/api v0.17.1
+	k8s.io/apimachinery v0.17.1
+	k8s.io/client-go v12.0.0+incompatible
+	k8s.io/klog v1.0.0
+	k8s.io/kube-openapi v0.0.0-20191107075043-30be4d16710a // indirect
+	k8s.io/utils v0.0.0-20191114200735-6ca3b61696b6
+	sigs.k8s.io/cluster-api-provider-aws v0.4.7
+	sigs.k8s.io/cluster-api-provider-azure v0.2.2
+	sigs.k8s.io/cluster-api-provider-openstack v0.2.0
+)
+
+replace (
+	github.com/Sirupsen/logrus => github.com/sirupsen/logrus v1.3.0 // Pulled in by tf-vsphere-provider
+	github.com/coreos/go-systemd => github.com/coreos/go-systemd v0.0.0-20181031085051-9002847aa142 // Pin non-versioned import v22.0.0
+	github.com/go-log/log => github.com/go-log/log v0.1.1-0.20181211034820-a514cf01a3eb // Pinned by MCO
```

Comment:
```
go mod why -m -vendor github.com/go-log/log
go: finding github.com/go-log/log v0.1.1-0.20181211034820-a514cf01a3eb
warning: ignoring symlink /home/adahiya/go/src/github.com/openshift/installer/pkg/asset/store/data
# github.com/go-log/log
(main module does not need to vendor module github.com/go-log/log)
```

hmm, this is replace required?

Reply:
> > [ccdd859](https://github.com/openshift/installer/commit/ccdd8592931264a0b7d9cd2b35763c1293ecef1e)
> 
> I see most of the replaces are Pinned or Replaced by MCO .. ? can you provide a little more context on that.. i'm not sure how replace is working here.. Thanks!

If I remove this, this happens:
```
[lorbus@localhost installer]$ go mod tidy
warning: ignoring symlink /home/lorbus/code/installer/pkg/asset/store/data
go: github.com/openshift/machine-config-operator@v4.2.0-alpha.0.0.20190917115525-033375cbe820+incompatible requires
        github.com/go-log/log@v0.0.0-00010101000000-000000000000: invalid version: unknown revision 000000000000
```

go mod pulls in the require directives of its deps, but not their replace directives (in this case: https://github.com/openshift/machine-config-operator/blob/master/go.mod#L75)